run.sh

```diff
@@ -20,29 +20,24 @@ datecmd() {
   date --utc $@
 }
 
-[[ -z "${AWS_ACCESS_KEY_ID+x}" ]] &&
-  fail "You need to set the AWS_ACCESS_KEY_ID environment variable."
-
-[[ -z "${AWS_SECRET_ACCESS_KEY+x}" ]] &&
-  fail "You need to set the AWS_SECRET_ACCESS_KEY environment variable."
-
-[[ -z "${AWS_BUCKET+x}" ]] &&
-  fail "You need to set the AWS_BUCKET environment variable."
-
-[[ -z "${PREFIX+x}" ]] &&
-  fail "You need to set the PREFIX environment variable."
-
-[[ -z "${MYSQL_ENV_MYSQL_USER+x}" ]] &&
-  fail "You need to set the MYSQL_ENV_MYSQL_USER environment variable."
-
-[[ -z "${MYSQL_ENV_MYSQL_PASSWORD+x}" ]] &&
-  fail "You need to set the MYSQL_ENV_MYSQL_PASSWORD environment variable."
-
-[[ -z "${MYSQL_PORT_3306_TCP_ADDR+x}" ]] &&
-  fail "You need to set the MYSQL_PORT_3306_TCP_ADDR environment variable or link to a container named MYSQL."
-
-[[ -z "${MYSQL_PORT_3306_TCP_PORT+x}" ]] &&
-  fail "You need to set the MYSQL_PORT_3306_TCP_PORT environment variable or link to a container named MYSQL."
+declare -A req_vars
+
+AWS_BUCKET=foo
+
+req_vars=(
+  [AWS_ACCESS_KEY_ID]='You need to set the AWS_ACCESS_KEY_ID environment variable.'
+  [AWS_SECRET_ACCESS_KEY]='You need to set the AWS_SECRET_ACCESS_KEY environment variable.'
+  [AWS_BUCKET]='You need to set the AWS_BUCKET environment variable.'
+  [PREFIX]='You need to set the PREFIX environment variable.'
+  [MYSQL_ENV_MYSQL_USER]='You need to set the MYSQL_ENV_MYSQL_USER environment variable.'
+  [MYSQL_ENV_MYSQL_PASSWORD]='You need to set the MYSQL_ENV_MYSQL_PASSWORD environment variable.'
+  [MYSQL_PORT_3306_TCP_ADDR]='You need to set the MYSQL_PORT_3306_TCP_ADDR environment variable or link to a container named MYSQL.'
+  [MYSQL_PORT_3306_TCP_PORT]='You need to set the MYSQL_PORT_3306_TCP_PORT environment variable or link to a container named MYSQL.'
+)
+
+for k in "${!req_vars[@]}"; do
```

Comment:
I agree that associative arrays aren't a common bash feature but I think what is going on here is fairly obvious.  It is certainly DRYer than cut'n'pasting a test for each variable.